bcda/web/api.go

```diff
@@ -213,6 +231,33 @@ func check429(jobs []models.Job, types []string, w http.ResponseWriter) ([]strin
 	}
 }
 
+func validateRequest (r *http.Request) ([]string, *fhirmodels.OperationOutcome)  {
+	var resourceTypes []string
+	params, ok := r.URL.Query()["_type"]
+	if ok {
+		resourceMap := make(map[string]bool)
+		params = strings.Split(params[0], ",")
+		for _, p := range params {
+			if p != "ExplanationOfBenefit" && p != "Patient" && p != "Coverage" {
+				oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, "Invalid resource type", responseutils.RequestErr)
+				return nil, oo
+			} else {
+				if !resourceMap[p] {
+					resourceMap[p] = true
+					resourceTypes = append(resourceTypes, p)
+				} else {
+					oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, "Repeated resource type", responseutils.RequestErr)
+					return nil, oo
+				}
+			}
+		}
+	} else {
+		// patient all request, all resource types applied
```

Comment:
this might not be a `/Patient` request.  the new `/Group` comes through this path too, right?
maybe this comment could be updated to say something like:
```
No resource types supplied in request; default to applying all resource types.
```

Reply:
Matt with the eyes!!